pkg/lib/cockpit/_internal/parentwebsocket.ts

```diff
@@ -11,9 +11,19 @@ import { transport_origin } from './location';
  *  * An empty string message means "close" (not completely used yet)
  */
 export class ParentWebSocket {
+    binaryType = 'arraybuffer' as const; // compatibility with Transport, which sets this
     readyState = 0;
 
-    constructor(parent) {
+    onopen(): void {
```

Comment:
What do these empty methods do? ParentWebSocket isn't derived from any class. This needs to be explained in a comment at least. If it's part of the API for purely TypeScript checking reasons, perhaps they should throw some "not implemented" error?

Reply:
Ya, it's a bit of an interesting case: basically, we're emulating the API of WebSocket, and this is part of that API.  If we don't have these, TS will complain when we assign to them in the `Transport` code.

The `throw new Error('NotImplementedError')` thing seems a bit Pythony, but maybe that's OK.  The reason I'd avoid it, though, is more to do with the fact that, strictly speaking, it's OK to have those unset (in which case, nothing happens).

I can add a comment.

Reply:
I meant, we are not really expecting onopen() and friends to be called, right? i.e. these should probably be more like `new Error("hahahno!")`, err, I mean "internal program error"? Not important, a simple comment will do as well.

Reply:
I did the comment.  Maybe someone will someday want to use this from somewhere other than the transport code!  :laughing: